### ISSUE TITLE
amplify project with "-" in name causes resource naming issues

### ISSUE THREAD

**Issue by julien-tamade** [aws-amplify/amplify-cli#4156]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
I created a project that had a hyphen in the name. When I created a storage resource I noticed the bucket it created didn't follow the conventional formula of bucketNamexxxxx-env, but instead was bucketNameenv-env. Looking into the cloudformation template, it's configuring the bucket name as 

```
"Fn::Select": [
      3,
       {
             "Fn::Split": [
                    "-",
                       {
                                "Ref": "AWS::StackName"
                        }
              ]
       }
 ]
```

normally this would pick up the unique number combo for the stackname, but because my stack name had an extra hyphen, it picked up the environment name instead. I'm not sure if this will affect other resources, but I decided to delete my project and start over so i don't run into issues in the future. 

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system
4.18.9

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable
create a project with a hyphenated name. and then add storage and push.

**Expected behavior**
I would expect the generated bucket to follow the standard naming convention

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac, Catalina
 - Node Version: 12.16.1

**Additional context**
Obviously, this isn't a huge issue. I think a warning would suffice when naming a project or maybe storing a reference to that number string instead of getting it by splitting a string based on the hyphens.

**Comment by jhockett** [comment]
Hey @julien-tamade can you check the CLI version you are using? The most up-to-date release at this moment is 4.18.1. If that was just a typo, and you're on 4.18.0, please update to the newest version. We fixed this issue in the latest release with #3893.

**Comment by julien-tamade** [comment]
I was on 4.18.0, so you can disregard. thanks anyway!
